python/beeai_framework/agents/experimental/agent.py

```diff
@@ -98,22 +102,31 @@ def __init__(
         self._tools = list(tools or [])
         self._requirements = list(requirements or [])
         self._meta = AgentMeta(name=name or "", description=description or "", tools=self._tools)
-        self.middlewares.extend(middlewares or [])
 
-    def run(
-        self,
-        prompt: str | None = None,
-        *,
-        context: str | None = None,
-        expected_output: str | type[TOutput] | None = None,
-        execution: AgentExecutionConfig | None = None,
-        signal: AbortSignal | None = None,
-    ) -> Run[RequirementAgentRunOutput[TOutput]]:
-        run_config = execution or AgentExecutionConfig(
-            max_retries_per_step=3,
-            total_max_retries=3,
-            max_iterations=20,
+    @runnable_entry
+    async def run(self, input: str | list[AnyMessage], /, **kwargs: Unpack[AgentOptions]) -> RequirementAgentOutput:
+        """Execute the agent.
+
+        Args:
+            input: The input to the agent (if list of messages, uses the last message as input)
+            expected_output: Pydantic model or instruction for steering the agent towards an expected output format.
+            total_max_retries: Maximum number of model retries.
+            max_retries_per_step: Maximum number of model retries per step.
+            max_iterations: Maximum number of iterations.
+            backstory: Additional piece of information or background for the agent.
+            signal: The agent abort signal
+            context: A dictionary that can be used to pass additional context to the agent
+
+        Returns:
+            The agent output.
```

Comment:
Duplicated as everything has already been defined in `AgentOptions`.

Reply:
I was just trying to document the kwargs that this specific agent uses. Not all agents use all of the kwargs currently. 
Let's fix and normalize the docstrings in our next iteration. This one has become a bit too long already, and I feel like once we bring the agents together, a lot of this is going to become cleaner and simpler. :)